test/unit/gateways/braintree_blue_test.rb

```diff
@@ -473,6 +473,42 @@ def test_store_with_existing_customer_id
     assert_equal 'cctoken', response.params['credit_card_token']
   end
 
+  def test_store_with_existing_customer_id_and_nil_billing_address_options
+    credit_card = stub(
+      customer_id: 'customerid',
+      token: 'cctoken'
+    )
+    options = {
+      :customer => 'customerid',
+      :billing_address => {
+        :name => 'John Smith',
+        :phone => '123-456-7890',
+        :company => nil,
+        :address1 => nil,
+        :address2 => nil,
+        :city => nil,
+        :state => nil,
+        :zip => nil,
+        :country_name => nil
+      }
+    }
+
+    result = Braintree::SuccessfulResult.new(credit_card: credit_card)
+    Braintree::CustomerGateway.any_instance.expects(:find).with('customerid')
+    Braintree::CreditCardGateway.any_instance.expects(:create).with do |params|
```

Comment:
I had no idea this was a thing you could do. Awesome!

Reply:
Er, this might be specific to the Braintree gem?